### DISCUSSION TITLE
Pinning, Vertex Colors and Skelatal Meshes

### DISCUSSION THREAD

**Discussion by NHodgesVFX** [ManevilleF/bevy_silk#2] [Ideas]
Trying to figure out what vertex is what when you have a character with 50, 000 of them is not easy. I propose adding the ability to use mesh vertex colors to provide which vertices to pin. 

A simple example would be any thing white is pinned and anything is black is not.

Another issue is if you want to use cloth on a animated skeletal mesh you need to pin to a bone or to the vertex of a skinned result instead of the global transform. For characters collisions are another issue,, just like with pinning you need the ability to attach the collision object to a bone.

Thanks for making this plugin :)

**Comment by ManevilleF** [comment]
Thanks for the feedback !

About vertex colors:

I was wondering how to handle fixed points in better ways, vertex color is a great idea !

Aboud skinned meshes:

I guess you are talking about clothes and capes, I need to look into the skinned mesh code to find if there is a way to retrieve such data. I think there could be a workaround with moving the cloth transform directly to follow the bone

**Comment by ManevilleF** [comment]
@NHodgesVFX https://github.com/ManevilleF/bevy_silk/pull/3 Adds vertex color pinning and other improvements.

But until bevy releases a new version vertex colors won't work as in bevy 0.7 vertex colors don't work

**Comment by ManevilleF** [comment]
Custom anchors are now available with #8 merged

**Comment by NHodgesVFX** [comment]
I think everything in this is actually possible now, I'll have to try adding cloth to a character once the collisions are stable. 

Thanks for adding all the features

**Reply by ManevilleF** [reply]
Collisions work quite well, you can try it out
